### PR TITLE
crosschain remote ed check, frozen assets handling

### PR DESCRIPTION
**Pull request by bnsports**

### CODE REVIEW

File: fearless/Modules/CrossChain/Model/CrossChainViewLoadingCollector.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        destinationFeeReady = false
+        balanceReady = false
+        existentialDepositReady = false
+        destinationBalanceReady = true
+        destinationExistentialDepositReady = false
+        assetAccountInfoReady = false
+    }
+
+    var isReady: Bool {
+        originFeeReady && destinationFeeReady && balanceReady && existentialDepositReady && destinationBalanceReady && destinationExistentialDepositReady && assetAccountInfoReady
```
Comment (DRadmir) [inline fearless/Modules/CrossChain/Model/CrossChainViewLoadingCollector.swift:23]:
```suggestion
                 [
            originFeeReady,
            destinationFeeReady,
            balanceReady,
            existentialDepositReady,
            destinationBalanceReady,
            destinationExistentialDepositReady,
            assetAccountInfoReady
         ].allSatisfy { $0 }
```

File: fearless/Modules/NewWallet/ChainAccount/ChainAccountPresenter.swift [hunk 2]
```diff
@@ -72,30 +73,40 @@ final class ChainAccountPresenter {
     }
 
     private func provideBalanceViewModel() {
-        var accountInfo: AccountInfo? = self.accountInfo
+        guard
+            case var .value(accountInfoValue) = accountInfo,
+            case let .value(frozenValue) = frozen,
+            case let .value(balanceValue) = balance,
+            case let .value(balanceLocksValue) = balanceLocks else {
```
Comment (DRadmir) [inline fearless/Modules/NewWallet/ChainAccount/ChainAccountPresenter.swift:80]:
```suggestion
            case let .value(balanceLocksValue) = balanceLocks 
            else {
```

File: fearless/Modules/CrossChainConfirmation/CrossChainDepsContainer.swift [hunk 3]
```diff
@@ -29,19 +31,37 @@ final class CrossChainDepsContainer {
...
+            ExistentialDepositService(
+                operationManager: OperationManagerFacade.sharedManager,
+                chainRegistry: chainRegistry,
+                chainId: $0
+            )
+        }
+        let storageRequestPerformer: StorageRequestPerformer? = (destChainModel?.chainId).flatMap {
+            guard
+                let runtimeService = chainRegistry.getRuntimeProvider(for: $0),
+                let connection = chainRegistry.getConnection(for: $0) else {
```
Comment (DRadmir) [inline fearless/Modules/CrossChainConfirmation/CrossChainDepsContainer.swift:54]:
```suggestion
                let connection = chainRegistry.getConnection(for: $0) 
                else {
```

File: fearless/Modules/NewWallet/ChainAccount/ChainAccountViewLayout.swift [hunk 4]
```diff
@@ -186,7 +186,7 @@ final class ChainAccountViewLayout: UIView {
     }
 
     func bind(balanceViewModel: ChainAccountBalanceViewModel?) {
-        balanceInfoStackView.isHidden = balanceViewModel == nil
+//        balanceInfoStackView.isHidden = balanceViewModel == nil
```
Comment (DRadmir) [inline fearless/Modules/NewWallet/ChainAccount/ChainAccountViewLayout.swift:189]: remove

**Review by DRadmir — DISMISSED** [review]

**Review by Viscount88 — APPROVED** [review]

**Review by DRadmir — APPROVED** [review]

**Pull request merged**